### ISSUE TITLE
Lab2 Review - Claudio Savelli (S317680)

### ISSUE THREAD

**Issue by ClaudioSavelli** [Mattizza/Computational_Intelligence_2023-2024#9]
Starting with the clarity of the proposed notebook, I find the theoretical description of the problem you faced, and what led you to make each choice during the development of the lab, extremely interesting and comprehensive. I also find very interesting the personal comments and criticisms that you made in the notebook, turning the lab into a kind of lecture. 
My only note to the code is that it would have been useful to comment on it more to make it easier and smoother to read, but after a first glance, it is easy to interpret and understand how it works. 

While it is true that Nim-Sum was used for fitness evaluation, which, as you also say, is not really legal considering that it leads to an exact solution, I very much appreciated how you exploited the latter to evaluate the opponent's best move within each iteration, not just analysing the state of the game but creating a sort of strategy that puts together an evolutionary algorithm and a sort of Minimax, playing considering the opponent's best possible moves. 

On the other hand, I found the graph extremely interesting and clear, which compares the wins obtained against a given opponent by changing the population size and, as expected, as the population size increases, the percentage of games won also increases. 
A question comes to mind regarding the last comment you made in the conclusion: 
"Remember that higher values do not always result in higher performance, and that there could be threshold value characteristic of the problem after which any increment may be detrimental."
in fact, over an infinite number of matches, shouldn't it be assured to get an improvement by increasing the population considering that the fitness is "perfect"? 

In conclusion, I find your work very valuable and interesting, a source of numerous insights!
